backend/src/index.js

```diff
@@ -433,24 +438,32 @@ async function main() {
     try {
       if (stakingNodes.length > 0) {
         for (let i = 0; i < stakingNodes.length; i++) {
-          const { account_id, currentStake: currentNodeStake } = stakingNodes[
-            i
-          ];
+          const {
+            account_id,
+            stakingStatus,
+            currentStake: activeNodeStake,
+          } = stakingNodes[i];
 
           try {
+            let currentStake = activeNodeStake;
             const account = await nearRpc.query({
               request_type: "view_account",
               account_id: account_id,
               finality: "final",
             });
 
-            const currentStake =
-              currentNodeStake ??
-              (await nearRpc.callViewMethod(
+            // query and update 'currentStake' for 'on-hold', 'newcomer' and 'idle' nodes
+            // because other nodes receive 'currentStake' from 'queryEpochStats()'
+            if (
+              !stakingStatus ||
+              nonValidatingNodeStatuses.indexOf(stakingStatus) >= 0
+            ) {
```

Comment:
We need it because when validator's stake decreased or increased we must to check is this validator is still has "idle" status? Maybe his status is "on-hold" now? And vice versa. Earlier we have checked this only at the first time